Source/SwiftLintFramework/Rules/Lint/CommentSpacingRule.swift

```diff
@@ -0,0 +1,143 @@
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "comment_spacing",
+        name: "Comment Spacing",
+        description: "Prefer at least one space after slashes for comments.",
+        kind: .lint,
+        nonTriggeringExamples: [
+            Example("""
+            // This is a comment
+            """),
+            Example("""
+            /// Triple slash comment
+            """),
+            Example("""
+            // Multiline double-slash
+            // comment
+            """),
+            Example("""
+            /// Multiline triple-slash
+            /// comment
+            """),
+            Example("""
+            /// Multiline triple-slash
+            ///   - This is indented
+            """),
+            Example("""
+            // - MARK: Mark comment
+            """),
+            Example("""
+            /* Asterisk comment */
+            """),
+            Example("""
+            /*
+                Multiline asterisk comment
+            */
+            """)
+        ],
+        triggeringExamples: [
+            Example("""
+            //↓Something
+            """),
+            Example("""
+            //↓MARK
+            """),
+            Example("""
+            //↓👨‍👨‍👦‍👦Something
+            """),
+            Example("""
+            func a() {
+                //↓This needs refactoring
+                print("Something")
+            }
+            //↓We should improve above function
+            """),
+            Example("""
+            ///↓This is a comment
+            """),
+            Example("""
+            /// Multiline triple-slash
+            ///↓This line is incorrect, though
+            """),
+            Example("""
+            //↓- MARK: Mark comment
+            """)
+        ],
+        corrections: [
+            Example("//↓Something"): Example("// Something"),
+            Example("//↓- MARK: Mark comment"): Example("// - MARK: Mark comment"),
+            Example("""
+            /// Multiline triple-slash
+            ///↓This line is incorrect, though
+            """): Example("""
+            /// Multiline triple-slash
+            /// This line is incorrect, though
+            """),
+            Example("""
+            func a() {
+                //↓This needs refactoring
+                print("Something")
+            }
+            //↓We should improve above function
+            """): Example("""
+            func a() {
+                // This needs refactoring
+                print("Something")
+            }
+            // We should improve above function
+            """)
+        ]
+    )
+
+    public func violationRanges(in file: SwiftLintFile) -> [NSRange] {
+        // Find all comment tokens in the file and regex search them for violations
+        let commentTokens = file.syntaxMap.tokens.filter { [.comment, .docComment].contains($0.kind) }
```

Comment:
Any reason not to include all comment kinds here?

```suggestion
        let commentTokens = file.syntaxMap.tokens.filter { SyntaxKind.commentKinds.contains($0.kind) }
```

Reply:
Ah great call, will definitely put this in